server/sonar-server/src/main/java/org/sonar/server/user/ws/SetHomepageAction.java

```diff
@@ -51,9 +73,47 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) throws Exception {
+    userSession.checkLoggedIn();
 
-    // WIP : Not implemented yet.
-    response.noContent();
+    String type = request.mandatoryParam(PARAM_TYPE);
+    String value = request.param(PARAM_VALUE);
+
+    if (PROJECT.toString().equals(type) && isBlank(value)) {
+      throw new IllegalArgumentException("type PROJECT requires a mandatory project key");
+    }
+
+    if (ORGANIZATION.toString().equals(type) && isBlank(value)) {
+      throw new IllegalArgumentException("type ORGANIZATION requires a mandatory project key");
+    }
+
+    String userLogin = userSession.getLogin();
+
+    String uuid = null;
 
+    try (DbSession dbSession = dbClient.openSession(false)) {
+
+      UserDto userDto = dbClient.userDao().selectActiveUserByLogin(dbSession, userLogin);
+      checkState(userDto != null, "User login '%s' cannot be found", userLogin);
+
+      if (PROJECT.toString().equals(type)) {
+        uuid = componentFinder.getByKey(dbSession, value).uuid();
+      }
+
+      if (ORGANIZATION.toString().equals(type)) {
+        uuid = checkFoundWithOptional(dbClient.organizationDao().selectByKey(dbSession, value), "No organizationDto with key '%s'", value).getUuid();
+      }
+
+      if ((MY_PROJECTS.toString().equals(type) || MY_ISSUES.toString().equals(type)) && value != null) {
+        throw new IllegalArgumentException("value parameter must be null when type = [MY_PROJECTS | MY_ISSUES]");
```

Comment:
Parameter value must not be provided when type is %s

Reply:
done